src/mmrelay/meshtastic/ble.py

```
@@ -0,0 +1,770 @@
+            bool: `True` if a device with the target BLE address is discovered within the timeout, `False` otherwise (including when BLE discovery errors occur).
+        """
+        try:
+            find_device = getattr(BleakScanner, "find_device_by_address", None)
+            if callable(find_device):
+                try:
+                    coro: Coroutine[Any, Any, Any] = cast(
+                        Coroutine[Any, Any, Any],
+                        find_device(ble_address, timeout=timeout),
+                    )
+                    result = await coro
+                    return result is not None
```

Ruff (0.15.9)

[warning] Consider moving this statement to an `else` block

(TRY300)